modules/programs/swayr.nix

```diff
@@ -0,0 +1,135 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.swayr;
+  tomlFormat = pkgs.formats.toml { };
+  configFile = tomlFormat.generate "config.toml" cfg.settings;
+  finalConfig = pkgs.writeText "swayr.toml"
+    ((builtins.readFile configFile) + cfg.extraConfig);
+in {
+  meta.maintainers = [ lib.hm.maintainers."9p4" ];
+
+  options.programs.swayr = {
+    enable = mkEnableOption "the swayr service";
+
+    settings = mkOption {
+      type = tomlFormat.type;
```

Comment:
I would prefer `type.nullOr tomlFormat.type;` so one can bypass nix handling of the config